### ISSUE TITLE
Dropped streams: SEVERE: [6327] XmppCommon.handleIQRequest#177: Exception handling IQ request

### ISSUE THREAD

**Issue by dpausp** [jitsi/jitsi-videobridge#1434]
## Description

Videobridge dropped all streams after a NullPointerException:

~~~
SEVERE: [6327] XmppCommon.handleIQRequest#177: Exception handling IQ request
java.lang.NullPointerException
        at org.jitsi.videobridge.shim.ChannelShim.<init>(ChannelShim.java:163)
        at org.jitsi.videobridge.shim.ContentShim.createRtpChannel(ContentShim.java:135)
        at org.jitsi.videobridge.shim.ContentShim.getOrCreateChannelShim(ContentShim.java:276)
        at org.jitsi.videobridge.shim.VideobridgeShim.processChannels(VideobridgeShim.java:141)
        at org.jitsi.videobridge.shim.VideobridgeShim.handleColibriConferenceIQ(VideobridgeShim.java:313)
        at org.jitsi.videobridge.Videobridge.handleColibriConferenceIQ(Videobridge.java:427)
        at org.jitsi.videobridge.xmpp.XmppCommon.handleIQRequest(XmppCommon.java:158)
        at org.jitsi.videobridge.xmpp.XmppCommon.handleIQInternal(XmppCommon.java:116)
        at org.jitsi.videobridge.xmpp.XmppCommon.handleIQ(XmppCommon.java:87)
        at org.jitsi.videobridge.xmpp.ClientConnectionImpl.handleIq(ClientConnectionImpl.java:97)
        at org.jitsi.xmpp.mucclient.IQListener.handleIq(IQListener.java:50)
        at org.jitsi.xmpp.mucclient.MucClient.handleIq(MucClient.java:566)
        at org.jitsi.xmpp.mucclient.MucClient.access$700(MucClient.java:50)
        at org.jitsi.xmpp.mucclient.MucClient$2.handleIQRequest(MucClient.java:530)
        at org.jivesoftware.smack.AbstractXMPPConnection$4.run(AbstractXMPPConnection.java:1188)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
INFO: [6335] [confId=79b1c76a4f765bf2 epId=716fdf7e gid=297217 stats_id=Minnie-MN2 conf_name=xxx] DtlsTransport.stop#184: Stopping
INFO: [6335] [confId=79b1c76a4f765bf2 epId=716fdf7e local_ufrag=1cne31ehseuelr gid=297217 stats_id=Minnie-MN2 conf_name=xxx] IceTransport.stop#235: Stopping
INFO: [6251] [confId=79b1c76a4f765bf2 gid=297217 stats_id=Minnie-MN2 componentId=1 conf_name=xxx ufrag=1cne31ehseuelr name=stream-716fdf7e epId=716fdf7e local_ufrag=1cne31ehseuelr] MergingDatagramSocket$SocketContainer.runInReaderThread#770: Failed to receive: java.net.SocketException: Socket closed
INFO: [6335] [confId=79b1c76a4f765bf2 gid=297217 stats_id=Minnie-MN2 componentId=1 conf_name=xxx ufrag=1cne31ehseuelr name=stream-716fdf7e epId=716fdf7e local_ufrag=1cne31ehseuelr] MergingDatagramSocket.close#142: Closing.
INFO: [6249] [confId=79b1c76a4f765bf2 epId=716fdf7e local_ufrag=1cne31ehseuelr gid=297217 stats_id=Minnie-MN2 conf_name=xxx] IceTransport.startReadingData#201: Socket closed, stopping reader
INFO: [6249] [confId=79b1c76a4f765bf2 epId=716fdf7e local_ufrag=1cne31ehseuelr gid=297217 stats_id=Minnie-MN2 conf_name=xxx] IceTransport.startReadingData#213: No longer running, stopped reading packets
INFO: [6253] [confId=79b1c76a4f765bf2 gid=297217 stats_id=Minnie-MN2 componentId=1 conf_name=xxx ufrag=1cne31ehseuelr name=stream-716fdf7e epId=716fdf7e local_ufrag=1cne31ehseuelr] MergingDatagramSocket$SocketContainer.runInReaderThread#770: Failed to receive: java.net.SocketException: Socket closed
INFO: [6257] [confId=79b1c76a4f765bf2 gid=297217 stats_id=Minnie-MN2 componentId=1 conf_name=xxx ufrag=1cne31ehseuelr name=stream-716fdf7e epId=716fdf7e local_ufrag=1cne31ehseuelr] MergingDatagramSocket$SocketContainer.runInReaderThread#770: Failed to receive: java.net.SocketException: Socket closed
INFO: [6335] [confId=79b1c76a4f765bf2 epId=716fdf7e gid=297217 stats_id=Minnie-MN2 conf_name=xxx] Endpoint.expire#778: Expired.
INFO: [6259] [confId=79b1c76a4f765bf2 epId=56eef282 gid=297217 stats_id=Wilford-cNf conf_name=xxx] TlsClientImpl.notifyAlertReceived#211: close_notify received, connection closing
~~~

## Steps to reproduce

Unclear, happened today for the first time. Didn't see any obvious external causes.

# Environment details

jitsi-videobridge2_2.1-304-g8488f77d-1_all.deb on NixOS, Kernel 5.4.55

openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-ga)
OpenJDK 64-Bit Server VM (build 25.212-bga, mixed mode)

**Comment by bbaldino** [comment]
Can you attach a full jvb log?

**Comment by bgrozev** [comment]
> jitsi-videobridge2_2.1-304

This is fixed in 309 (via jitsi/jitsi-videobridge#1403).
